### DISCUSSION TITLE
Python Altair Charts when rendering to PDF

### DISCUSSION THREAD

**Discussion by hathawayj** [quarto-dev/quarto-cli#916] [Q&A] [answered]
The Altair charts are rendered to HTML output without an issue.  My first try to create PDF output failed. I thought it might, as Altair is a bit of a pain to convert to static graphics.  Any guidance on how rendering to PDF might work for Altair charts with Quarto?

**Comment by jjallaire** [comment]
We will eventually create a generic mechanism for converting HTML/SVG to raster graphics for PDF output (probably not for a few months though). cc @cscheid

**Comment by hathawayj** [comment]
That is great.  It is a bit of a mess teaching Altair when asking the students to export their images to png.  Quarto will go a long way to making setup less painful.  Looking forward to the update.

**Comment by ganong123** [comment]
@jjallaire I'm having the same problem now that @hathawayj was having two years ago! 

Was there any resolution?

**Reply by mcanouil** [reply]
@ganong123 I suggest you open a new discussion and/or add a fully reproducible example.
For example, a complete Quarto document or a Git repository.
The goal is to make it as easy as possible for us to recreate your problem so that we can fix it: please help us help you! Thanks.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by MarekOzana** [reply, marked as answer]
I created new bug report with reproducible example:
https://github.com/quarto-dev/quarto-cli/issues/10571
Thank you for your time

**Reply by mcanouil** [reply]
I wrote a GitHub Discussion not an issue.

**Reply by MarekOzana** [reply]
I think it is an issue, one cannot use altair charts with Beamer or PDF docs. One needs to go back to matplotlib. 
Just to relate, like if Quarto worked for R standard plotting library but not for ggplot, then it would be big thing ;)

**Reply by mcanouil** [reply]
> Just to relate, like if Quarto worked for R standard plotting library but not for ggplot, then it would be big thing ;)

This isn’t about prioritising the issue. Sometimes, Python/Jupyter (like R/knitr) requires slightly different syntax to function correctly, which isn’t related to Quarto. Beginning with a discussion allows us to investigate and determine if it’s a Quarto bug. If it turns out not to be a bug, it prevents unnecessary “noise” in the issues.
